File: hello-world-node/tests/handler.test.js
const test = require("node:test");
const assert = require("node:assert/strict");
const { handler } = require("../index.js");

test("handler returns hello message", async () => {
  const response = await handler();
  assert.equal(response.statusCode, 200);
  assert.match(response.body, /Node/);
});

File: hello-world-java/src/test/java/com/example/HandlerTest.java
package com.example;

import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertEquals;
import org.junit.jupiter.api.Test;

class HandlerTest {
    @Test
    void returnsHelloMessage() {
        Handler handler = new Handler();
        var response = handler.handleRequest(Map.of(), null);
        assertEquals(200, response.get("statusCode"));
        assertEquals("{\"message\":\"Hello from Java Lambda\"}", response.get("body"));
    }
}
